File: TelegramBot.py
from ChatBot import *
import telebot

class TelegramBot(ChatBot):
    def __init__(self, credfile):
        super().__init__("telegram", credfile)
        self.token = open(credfile, 'r').read()[:-1]
        self.bot = telebot.TeleBot(self.token)
        self.register_commands()

    def handle_msg(self, userid, msg):
        self.service.handle_msg(userid, msg)

    def send_msg(self, userid, msg):
        self.bot.send_message(userid, msg)

    def register_commands(self):
        @self.bot.message_handler()
        def handle_msg(message):
            self.handle_msg(message.from_user.id, message.text)

    def run(self):
        self.bot.infinity_polling()

File: TelegramBot.js
const { Telegraf } = require('telegraf');
const { message } = require('telegraf/filters');
const creds = require("./creds.json")
const ChatBot = require("./ChatBot.js")

class TelegramBot extends ChatBot {

    constructor(name) {
        super("TELEGRAM_TOKEN", name)
        this.token = this.creds
        this.init_client()
    }

    init_client() {
        this.bot = new Telegraf(this.token);
        this.bot.owner = this;
        this.bot.on(message('text'), (ctx) => {
            this.handle_msg(ctx.message.chat.id, ctx.message.text);
        })
    }

    handle_msg(chatid, msg) {
        if (this.service != null) {
            this.service.handle_msg(this.name, chatid, msg)
        }
    }

    send_msg(chatid, msg) {
        this.bot.telegram.sendMessage(chatid, msg)
    }

    run() {
        this.bot.launch();
    }

}

module.exports = TelegramBot
